### ISSUE TITLE
Journal of Statistical Software

### ISSUE THREAD

**Issue by RobinHankin** [RobinHankin/hyper2#153]
Submitted the result of compiling `inst/jss_hyper3.Rnw` to JSS just now, it is submission #4832

Here is the email, sent to `hankin.robin@gmail.com`:

Dear Dr Hankin

Thank you for submitting your manuscript, “Generalized Plackett-Luce Likelihoods” to Journal of Statistical Software. Our editorial team will check your submission within the next days and get back to you if there are any immediate concerns. Our journal management system will allow you to track the progress of your submission by logging in to the journal web site:

Submission URL: https://www.jstatsoft.org/index.php/jss/authorDashboard/submission/4832 Username: hankin

At Journal of Statistical Software the editors provide a first review round for all submitted papers. We feel this provides richer feedback to authors and ensures that there is a higher chance of acceptance for papers that reach the formal reviewing process. As the Journal of Statistical Software is completely based on voluntary work it might take some until you get feedback regarding you submission.

If you have any questions, please do not hesitate to contact us.

All best,

Editorial Office

-------------------------------

Journal of Statistical Software http://www.jstatsoft.org/

**Comment by RobinHankin** [comment]
email from JSS:

JSS 4832:

Generalized Plackett-Luce Likelihoods

Robin K. S. Hankin


---------------------------------------------------------
For further instructions on JSS style requirements please see the Style Files (in particular section 2.1 Style Checklist of jss.pdf) and FAQ at http://www.jstatsoft.org/about/submissions.

For further examples please see RECENT JSS papers.
---------------------------------------------------------

From the editorial team:

o The manuscript currently lacks a thorough software overview allowing the reader to assess which software packages (for R but also other software environments) are available providing related functionality and indicating how the presented functionality provides additional functionality.

o The manuscript could benefit from some restructuring to better guide the reader on available methods and implementations, the statistical methodology covered and extended and the specific computational implementation made. It would seem beneficial to entangle these different aspects as different readers might have differnt foci of interest and could thus better select on which parts of the manuscript to focus. The sections also currently seem to be a mix between different methods and illustrative applications. Again a thorough overview on the statistical functionality provided by the package in a separate section might ease accessibility for readers. Disussing specific computational aspects of the implementation in a separate section is certainly also of interest.

o Note that in R many functions providing ML estimates do not only return point estimates but either an object where then in an additional step, e.g., the Hessian can be determined as well as further information provided using summary / print etc., or directly have already an object returned which contains more information on the fit including also the log-likelihood attained, the model fitted, etc.

o As an example, methods(class = "PLADMM") in the PlackettLuce package lists: anova, deviance, fitted, itempar, logLik, predict, print, summary, vcov. The anova/fitted/predict might be specific to the regression case but the others might be useful for hyper3 as well. Similarly for btmodel() in the psychotools package and the eba package while BradleyTerry2, BTLLasso, PLMIX, and prefmod provide fewer methods.

o The NAMESPACE contains

exportPattern("^[[:alpha:]]+")

Please avoid this and check if this can be removed or how it could be dropped.

It would seem that currently functions are exported which are not supposed to be directly called by the user, e.g., no argument checking. In this way they could be kept internal and documentation if desired could be provided with keyword internal to avoid cluttering the output of

help(package = "hyper2")


Manuscript style comments:

o Code should have enough spaces to facilitate reading.  Please include spaces before and after operators and after commas (unless spaces have syntactical meaning).

o The rule for capitalizing the starting letters of Figure, Section and Table is as follows: If you are referring to a particular figure/section/table then capitalize the first letter, otherwise use a lower-case first letter. For example, something shown in Section 4 vs. there are three sections in this paper.

o In all cases, code input/output must fit within the normal text width of the manuscript.  Thus, code input should have appropriate line breaks and code output should preferably be generated with a suitable width (or otherwise edited).

o As a reminder, please make sure that:
	- \proglang, \pkg and \code have been used for highlighting throughout the paper (including titles and references), except where explicitly escaped.


References:



o Please make sure that all software packages are \cite{}'d properly.
o All references should be in title style.
o See FAQ for specific reference instructions.


Code:

o Please make sure that the files needed to replicate all code/examples within the manuscript are included in a standalone replication script.

**Comment by RobinHankin** [comment]
On 15th April 2023, I get referees comments, in PDF format and some text comments too:

The manuscript describes functionality of the hyper2 package for Fitting and
testing variants of the Placket-Luce ranking model.

1.1 Recommendation
The manuscript may be publishable in the journal after revisions have been
made.

1.2 General comments
G1 The introduction should provide an overview of the contents of the manuscript.
In addition, the section headlines could be made more informative. Currently, they are:
2. "Some simple cases"
3. "Mann-Whitney test"
4. "Multiple competitors: javelin"
5. "Formula 1 motor racing: the constructors' championship"

Maybe the headlines could provide more of a summary of what the section
contains.

G2 There should be a comparison with other software. Although the author
states in the introduction that with hyper2 "It becomes possible to analyse
a wider range of likelihood functions than standard Plackett-Luce", it is
not clear to me which of the features described later on are unique to the
hyper2 package and which can also be obtained from other packages (like,
e.g., PlackettLuce).

G3 The manuscript should comment on the timing of the functions. For the
simple example on P. 4, the maxp(H) call takes more than 8 seconds on
my computer. This makes me wonder how timings would be for realistic
examples.

1.3 Specific comments
S1 I'm not claiming any expertise in English language, but occasionally, I found
the manuscript difficult to understand. For example, in the abstract it
says: "The package . . . furnishes idiom" Couldn't that simply be: "The
package . . . provides functionality"? On P. 13, it says "Here we use generalized
Plackett-Luce to assess the constructors' performance".  Isn't there
a "model" missing? (Plackett-Luce is just a name and cannot be used).

S2 Figure 2: For readability, maybe use xlab = expression(p[a]) and maybe
instead of a legend, write AAB, ABA, BAA next to the line in the color
of that line.

S3 P. 7: "https:github.com/RobinHankin/hyper2" should be
"https://github.com/RobinHankin/hyper2".

S4 P. 7: "Wolf" should be "Wolfe".

S5 P. 15, maxp(const2020, n=1): What does n do?

S6 P. 16, "we see strong evidence for a real decrease in the strength": I'd tone
that down. The model is almost certainly wrong, so it does not measure
the "real decrease." Also, power is unknown, so it is difficult to say what
"evidence" there is against H0. (See, e.g., Gelman, Regression an other
stories, Chapter 4: Time to unlearn what you thought you knew about
statistics. https://statmodeling.stat.columbia.edu/2022/01/27/regression-
and-other-stories-free-pdf/)

S7 The references are lacking their DOIs.


2 Review of software

The package includes many functions with a similar name. E.g., ordervec2supp,
ordervec2supp3, ordervec2supp3a, ordertable2supp, ordertable2supp3. Maybe
there could be some overview of which to use for what purpose.


Section editor comments:


The author proposes an extension of the alreadty existing hyper2 package.
This new S3 class allows for some extended likelihood specifications.

Even though this is just a code snippet and the hyper2 package was already
published elsewhere, the author should provide some introductory elaborations
on the Plackett-Luce model. Along these lines it would also be helpful to give
a quick overview of the hyper2 package, and mention other related packages. As it
is now it is difficult for a reader who's not super familiar with PL-models and
hyper2, respectively, to follow the paper. Anyway, I find the runners example 
as motivating example very illustrative, and actually like the examples throughout
the paper.

Some more aesthetic effort should be made in the code file (e.g., providing
comments). The code looks like a simple purl() extraction.

**Comment by RobinHankin** [comment]
email from JSS editorial office:

Dear Robin K.S. Hankin,

We have reached a decision regarding your submission to Journal of Statistical Software, #4832 “Generalized Plackett-Luce Likelihoods”.

After reviewing your work we are pleased to inform you that your manuscript entitled “Generalized Plackett-Luce Likelihoods” is acceptable for publication subject to satisfying the technical style and other publication requirements, in particular complying with JSS's reproducibility policy.

At this point, you can consider your paper provisionally accepted. This is not a formal guarantee of final publication. Please note that the final acceptance letter along with a DOI, for your manuscript will be provided only once your submission is scheduled for publication after having complied with all publication requirements.

As a next step, please upload the full sources for your submission as a single .zip (or .tar.gz) to https://www.jstatsoft.org/authorDashboard/submission/4832 It should contain:

The full sources of the latest version of the software. (Binary versions can be provided in addition.)

The .tex, .bib, and final .pdf for the manuscript. Please name all files jss4832.tex, jss4832.bib, jss4832.pdf, and so on. Also include all graphics in a Figures/ directory, uploaded as a compressed .zip or .tar.gz file.

Information on how to replicate the examples in the manuscript. (Typically, this is a self-contained standalone script file that loads/calls/sources the software package from (1). If data or other external files are needed, that are not yet provided with the software package, include these as well.) Please use subdirectories for Figures/ and Code/, and upload these subdirectories as a single .zip or .tar.gz file. All scripts should be in the subdirectory Code/, and all figures produced by the scripts, if they are written to file, should be written to Figures/. Please make sure only necessary files are included. That is, please do not include .aux, .log, etc. files and any unused files such as jss.cls, jss.bst, jsslogo.jpg, etc.

Note for R authors: If you have prepared your manuscript using Sweave, the files for replication of the examples in (3) can be produced by Sweave. The files for the Code/ directory can be produced by Tangle (possibly enhancing the comments). Also indicate in your submission that Sweave was used.

Please ensure that the manuscript complies with our style guide (see https://www.jstatsoft.org/style) and the supplementary material meets the reproducibility requirements (see https://www.jstatsoft.org/authors).

After we receive the post-processed material for your manuscript/software/replication materials, we will check its suitability and eventually add it to our copy-editing queue. Please note that due to various major changes in JSS over the last few years, the publication delay (between being added to the queue and final publication) is currently about one year. We apologize for any inconveniences caused by this delay and ask for your patience. We try our best to reduce the delay.

Thanks for choosing JSS and contributing to free statistical software.

JSS Administrator<br/>[editor@jstatsoft.org](mailto:editor@jstatsoft.org)


-----------

**Comment by RobinHankin** [comment]
JSS 4832:

Generalized Plackett-Luce Likelihoods

Robin K. S. Hankin

---------------------------------------------------------
For further instructions on JSS style requirements please see the Style Files (in particular section 2.1 Style Checklist of jss.pdf) and FAQ at http://www.jstatsoft.org/about/submissions.

For further examples please see RECENT JSS papers.
---------------------------------------------------------

From the editorial team:

I had a closer look at the revision of #4832 and, as far as I'm concerned, the paper can be accepted as it is. However, I wanted to draw your attention to the author's rebuttal file, p.7. It seems that he doesn't agree that this paper should be published as code snippet, as opposed to a regular paper. I'm not sure whether this was previously communicated to the author by the editorial team. Anyway, I think it would be good if the editorial team could clarify this issue directly with the author.

Also, I think he misunderstood my concerns about the supplemental R code file (see bottom of p.7 in the rebuttal file). What I'm asking for is to make the R code look nicer with proper comments, as opposed to just pull it from the Rnw file. In addition, the code starts with

source("loadlibrary.R",echo=FALSE) # to avoid stupid messages
opts_chunk$set(cache=TRUE, autodep=TRUE)

This is not reproducible. Also, the author doesn't even load the hyper2 package.

So, while I think the paper is in good shape, the code file is not. But this can be easily be fixed.




Manuscript style comments:

o Please use "R>" as the command prompt, rather than ">".

o The code presented in the manuscript should not contain comments within the verbatim code. Instead the comments should be made in the normal LaTeX text.

o For the code layout in R publications, we typically distinguish input/output using Sinput/Soutput (or equivalently CodeInput/CodeOutput). Unless there are special reasons to format it differently, the input should use the text width (up to 76 or 77 characters) and be indented by two spaces, e.g.,

begin{Sinput}
R> example_model <- lm(response ~ variable1 + variable2 + variable3, 
+    weights = w, data = mydata)
\end{Sinput}

o As a reminder, please make sure that:
	- \proglang, \pkg and \code have been used for highlighting throughout the paper (including titles and references), except where explicitly escaped.


References:


o MATLAB (not: Matlab, matlab)
	guages. Hunter (2004), for example, presents results in matlab [although he works with a


o As a reminder,
	- Please make sure that all software packages are \cite{}'d properly.

	- All references should be in title style.

	- See FAQ for specific reference instructions.


Code:

o As a reminder, please make sure that the files needed to replicate all code/examples within the manuscript are included in a standalone replication script.

**Comment by RobinHankin** [comment]
Dear Dr Hankin,

It looks like the .zip file doesn't contain the software. To emphasize, the requirements in our last email about the .zip file are as follows.

It should contain:

The full sources of the latest version of the software. (Binary versions can be provided in addition.)

The .tex, .bib, and final .pdf for the manuscript. Please name all files jss4832.tex, jss4832.bib, jss4832.pdf, and so on. Also include all graphics in a Figures/ directory, uploaded as a compressed .zip or .tar.gz file.

Information on how to replicate the examples in the manuscript. (Typically, this is a self-contained standalone script file that loads/calls/sources the software package from (1). If data or other external files are needed, that are not yet provided with the software package, include these as well.) Please use subdirectories for Figures/ and Code/, and upload these subdirectories as a single .zip or .tar.gz file. All scripts should be in the subdirectory Code/, and all figures produced by the scripts, if they are written to file, should be written to Figures/. Please make sure only necessary files are included. That is, please do not include .aux, .log, etc. files and any unused files such as jss.cls, jss.bst, jsslogo.jpg, etc.
Best regards,

JSS editorial team

**Comment by RobinHankin** [comment]
Dear Robin K.S. Hankin,

We have reached a decision regarding your submission to Journal of Statistical Software, #4832 “Generalized Plackett-Luce Likelihoods”.

After reviewing your work I am pleased to inform you that your manuscript entitled “Generalized Plackett-Luce Likelihoods” is acceptable for publication subject to satisfying the technical style and other publication requirements. Your manuscript will now be forwarded to the Technical Editor who will contact you in due course.

At this point, you can consider your paper provisionally accepted. This is not a formal guarantee of final publication. Please note that the final acceptance letter along with a DOI, for your manuscript will be provided only after the Technical Editor approves its publication.

Please respond promptly to requests from the Technical Editor to ensure a speedy route to final publication. Thank you for choosing the Journal of Statistical Software as your publication outlet.

Yours sincerely,

The editorial team

JSS Administrator<br/>[editor@jstatsoft.org](mailto:editor@jstatsoft.org)

**Comment by RobinHankin** [comment]
Email from the JSS editorial team:

```
Dear Dr Hankin,

We are nearly ready to prepare your manuscript "Generalized Plackett-Luce Likelihoods" for publication in the Journal of Statistical Software. The current version of your submission has been attached, incorporating editorial updates. A detailed overview of all changes can be found in the README file within the attached zip file.

There are a few outstanding issues we wanted to resolve before the final publication. They are listed as follows:

1. Please add the CRAN-URL to the manuscript similar to:
"\code{hyper3} is provided in the package \pkg{hyper2}\citep{pkg:hyper2}, available from the Comprehensive \proglang{R} Archive Network (CRAN) at \url{https://CRAN.R-project.org/hyper2}."
2. If you have an ORCID, please include with:
"\author{Robin K. S. Hankin~\orcidlink{<orcid-link>}}"
3. There appears to be an issue with the URL in your affiliation as the corresponding website does not seem to exist. Please verify this.
4. Please ensure section titles are concise enough to fit on one line.
5. Please confirm that all code/package names/classes are correctly highlighted using:
- \pkg{<package-name>} for package names
- \code{<code>} for code/functions
- \code{<class>} for classes
6. We could not exactly reproduce the results on page 15, e.g. for "null estimate: RBRH", please check (see our results in Code/code.Rout)

Please use the attached .tex file for any further edits. We ask that you provide the requested updates within the next two weeks. If this timeline is not feasible, kindly inform us of a more suitable timeframe.

Should you have any questions or need further assistance, do not hesitate to contact me.

Best regards,

Sandra
```

**Comment by RobinHankin** [comment]
OK the general form to address Sandra's concerns 1-6 above will be:
1. `git add paper.tex ref.bib` to the repo (these files being part of file `4832.zip` which I downloaded from the JSS editorial manager)
2. work on 1-6 above, editing `paper.tex` [mentioning issue #153 in the commit]
3. when happy, upload `paper.tex` and `ref.bib` 
4. if  `ref.bib` does need to be edited, incorporate the changes to `vignettes/hyper2.bib`
5. rename `paper.tex` to `jss4832.tex` 
6. close this issue

**Comment by RobinHankin** [comment]
bullet point 5 requires no action

**Comment by RobinHankin** [comment]
email from Achim just now:

Robin,

we prepared your accepted manuscript for publication in the next JSS volume. Please find attached the PDF manuscript that contains various editorial touch-ups. Specifically, we added ORCIDs where available as well as DOIs for many of the references, please check whether these are correct (if you haven't already done so in the copyediting stage). Moreover, we have omitted the options() calls in the paper as they are in the replication code.

One open question remains, namely: The paper mentions a discussion in "inst/fisher_inf_PL3.Rmd." However, it is unclear to us where to find this file/discussion.

Finally, please release a new version of the hyper2 package to CRAN that refers back to the JSS manuscript. To do so, please include the attached CITATION file as hyper2/inst/CITATION (without .txt or any other suffix). Also include the corresponding reference in the DESCRIPTION/Description as <doi:10.18637/jss.v109.i08> and in the \references{} sections of the relevant .Rd pages.

Please let me know whether you will be able to make all of these updates within the next 7 days. If not, this is no problem but it might further delay the publication of your manuscript. Due to this the DOI currently assigned to your manuscript has not been registered yet. Hence, when uploading your package to CRAN, please add the following comment: "The DOI in the CITATION is for a new JSS publication that will be registered after publication on CRAN."

Thanks in advance & best regards,
Achim

**Comment by RobinHankin** [comment]
email from Achim:

Robin, I just noticed that it would be good to merge the new citation with the existing one. So I attach a combined CITATION that you can include in the package.

CITATION	zeileis
2024-05-29 11:47 PM

My reply:

Thanks for this Achim, I'm working on it and should be able to upload everything over the next few days.  Thank you for the new CITATION file.  I've been using the 2017 R Journal article as the default reference for this work and I've been wondering how to maintain that with the generalized methods.

Best wishes

Robin

**Comment by RobinHankin** [comment]
uploaded `hyper2_3.1.-0.tar.gz` to `CRAN` just now.

**Comment by RobinHankin** [comment]
My reply 3 June 2024:

thanks for this Achim, sorry for the delay.  I uploaded  hyper2_3.1-0.tar.gz to CRAN on Friday, as per your intstructions. For the manuscript, all looks good except the following bullet points:

1.  the reference to  "inst/fisher_inf_PL3.Rmd".   This file was renamed on the repo to the more descriptive filename "three_runners_plackett_luce.Rmd",and I didn't change the manuscript to reflect this change.  Sorry about that, please find the file attached  below.  I am not sure if it would be easier to revert the filename to "fisher_inf_PL3.Rmd" or to change the manuscript; your call.

2.  DOIs look good with the exception of the DOI for Turner and Firth 2012 which is incorrectly stated in the bib file, please find attached "turner_and_firt2012_corrected.bib" which has the correct DOI.

3. on page 7 of v09i08.pdf there is a slight typo, there is an open bracket before "thus", and there is no corresponding close bracket.  OLD TEXT:  "failed to finish.".  NEW TEXT:  "failed to finish)."

Best wishes 

 

Robin

**Comment by RobinHankin** [comment]
Robin, great, thanks for the thorough checks. I wrapped up everything and published it at doi:10.18637/jss.v109.i08

Thank you for all your contributions to JSS - and to free statistical software more generally!

Best wishes, Achim
